enhancements/authentication/allow-users-to-manage-their-own-tokens.md

```diff
@@ -0,0 +1,153 @@
+---
+title: neat-enhancement-idea
+authors:
+  - "@stlaz"
+reviewers:
+  - "@sttts"
+  - "@deads2k"
+  - "@marun"
+  - "@vareti"
+approvers:
+  - "@sttts"
+  - "@deads2k"
+creation-date: 2020-10-22
+last-updated: 2020-10-22
+status: implementable
+see-also:
+- [enhancement: Secure OAuth Resource Storage](https://github.com/openshift/enhancements/pull/323)
+replaces:
+superseded-by:
+
+---
+
+# Allow Users To Manage Their Own Tokens
+
+## Release Signoff Checklist
+
+- [x] Enhancement is `implementable`
+- [ ] Design details are appropriately documented from clear requirements
+- [ ] Test plan is defined
+- [ ] Graduation criteria for dev preview, tech preview, GA
+- [ ] User-facing documentation is created in [openshift-docs](https://github.com/openshift/openshift-docs/)
+
+## Summary
+
+While users are quite capable of retrieving access tokens to be able to
+work with OpenShift services, their capabilities to see the tokens they
+created and to handle them is a bit lacking. This enhancement describes
+how the self-management of users' access tokens can be improved.
+
+## Motivation
+
+It is quite easy for users to retrieve tokens but it is also not so hard
+for them to be able to lose access to them - people don't usually search
+their cookies for the value of the token, or don't keep history of the
+tokens they retrieve by issuing the `oc login` command multiple times.
+
+### Goals
+
+Allow users to:
+- list their own tokens
+- view details of a specific token
+- delete their own tokens
+
```

Comment:
would add: via console, some special `oc logout --all` command and via normal `oc get/delete` command line

Reply:
I am not sure. `logout` might suggest a feeling of logging out of sessions, but we don't touch anything but the tokens.

If we made the command remove only the tokens issued for the `openshift-challenging-client` and thus effectively clearing all CLI sessions, it might be confusing as to why not all the tokens were removed.

Reply:
agree about logout.